internal/services/oracle/oracle_exascale_database_virtual_machine_cluster_resource.go

```diff
@@ -0,0 +1,622 @@
+
+			if model := resp.Model; model != nil {
+				state.Location = model.Location
+				state.Tags = pointer.From(model.Tags)
+				state.Zones = pointer.From(model.Zones)
+
+				if props := model.Properties; props != nil {
+					state.DisplayName = props.DisplayName
+					state.EnabledEcpuCount = props.EnabledEcpuCount
+					state.ExascaleDbStorageVaultId = props.ExascaleDbStorageVaultId
+					state.GridImageOcid = pointer.From(props.GridImageOcid)
+					state.Hostname = props.Hostname
+					state.NodeCount = props.NodeCount
+					state.Shape = props.Shape
+					state.SshPublicKeys = props.SshPublicKeys
+					tmp := make([]string, 0)
+					for _, key := range props.SshPublicKeys {
+						if key != "" {
+							tmp = append(tmp, key)
+						}
+					}
+					state.SshPublicKeys = tmp
+					state.SubnetId = props.SubnetId
+					state.TotalEcpuCount = props.TotalEcpuCount
+					state.VirtualMachineFileSystemStorage = FlattenVMFileSystemStorage(props.VMFileSystemStorage)
+					state.VnetId = props.VnetId
+					state.Location = model.Location
+					state.Tags = pointer.From(model.Tags)
+					state.Zones = pointer.From(model.Zones)
+
+					state.BackupSubnetCidr = pointer.From(props.BackupSubnetCidr)
+					state.ClusterName = pointer.From(props.ClusterName)
+					state.DataCollectionOption = flattenExadbDataCollectionOptionInterface(metadata.ResourceData.Get("data_collection_option").([]interface{}))
+					state.Domain = pointer.From(props.Domain)
+					state.LicenseModel = string(pointer.From(props.LicenseModel))
+					state.NetworkSecurityGroupCidrs = FlattenNetworkSecurityGroupCidr(props.NsgCidrs)
+					state.Ocid = pointer.From(props.Ocid)
+					state.PrivateZoneOcid = pointer.From(props.PrivateZoneOcid)
+					state.SingleClientAccessNameListenerPortTcp = pointer.From(props.ScanListenerPortTcp)
+					state.SingleClientAccessNameListenerPortTcpSsl = pointer.From(props.ScanListenerPortTcpSsl)
+					state.SystemVersion = metadata.ResourceData.Get("system_version").(string)
+					state.TimeZone = pointer.From(props.TimeZone)
+					state.ZoneOcid = pointer.From(props.ZoneOcid)
+				}
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (ExascaleDatabaseVirtualMachineClusterResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Oracle.OracleClient.ExadbVMClusters
+
+			id, err := exadbvmclusters.ParseExadbVMClusterID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			if err = client.DeleteThenPoll(ctx, *id); err != nil {
+				return fmt.Errorf("deleting %s: %+v", *id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (ExascaleDatabaseVirtualMachineClusterResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return exadbvmclusters.ValidateExadbVMClusterID
+}
+
+func flattenExadbDataCollectionOptionInterface(input []interface{}) []ExascaleDatabaseDataCollectionOptionModel {
```

Comment:
This isn't flattening, this is setting the raw state to a decoded state, which can be done in another way.